### PR TITLE
fix(schema): Use validate tags instead of json omitempty

### PR DESCRIPTION
**Pull request by clemenskohlhs**
This changes the schema generation to listen to the `validation` tag instead of the `json` tag. This is needed since we don't want to omit values like `0` and `""` from the response of the API, but still want to validate the requests beforehand.

[Issue](https://github.com/heimspiel/go-dragon-utils/issues/75)

### CODE REVIEW

File: examples/chiexample/go.mod [hunk 1]
```diff
@@ -1,27 +1,27 @@
 module github.com/heimspiel/rest/examples/chiexample
 
-go 1.21
+go 1.22.0
```
Comment (jafhueppe) [inline examples/chiexample/go.mod:3]: suggestion: update to latest version (1.23)

File: schema.go [hunk 2]
```diff
@@ -483,8 +483,10 @@ func (api *API) RegisterModel(model Model, opts ...ModelOpts) (name string, sche
 				fieldName = f.Name
 			}
 
+			validationTags := strings.Split(f.Tag.Get("validate"), ",")
+
 			tempOpts := []ModelOpts{}
-			if len(jsonTags) > 1 && jsonTags[1] == "omitempty" {
+			if slices.Contains(validationTags, "omitempty") {
```
Comment (jafhueppe) [inline schema.go]: suggestion: can't we just check the string returned by `f.Tag.Get()` if it contains "omitempty"? I can't see another usage of `validationTags`, so it shouldn't hurt?

File: schema.go [hunk 3]
```diff
@@ -518,8 +520,11 @@ func (api *API) RegisterModel(model Model, opts ...ModelOpts) (name string, sche
 			}
 			schema.Properties[fieldName] = ref
 			isPtr := fieldType.Kind() == reflect.Pointer
-			hasOmitEmptySet := slices.Contains(jsonTags, "omitempty")
-			if isFieldRequired(isPtr, hasOmitEmptySet) {
+
+			// This removes the need for the omitempty tag, to make the field non required.
+			// Required fields should be provided through a CustomSchemaApplier (https://pkg.go.dev/github.com/a-h/rest#CustomSchemaApplier)
+			//hasOmitEmptySet := slices.Contains(jsonTags, "omitempty")
+			if isFieldRequired(isPtr, true) {
```
Comment (jafhueppe) [inline schema.go]: question: to be sure - this `true` is intendend here?

File: schema.go [hunk 4]
```diff
@@ -518,8 +520,11 @@ func (api *API) RegisterModel(model Model, opts ...ModelOpts) (name string, sche
 			}
 			schema.Properties[fieldName] = ref
 			isPtr := fieldType.Kind() == reflect.Pointer
-			hasOmitEmptySet := slices.Contains(jsonTags, "omitempty")
-			if isFieldRequired(isPtr, hasOmitEmptySet) {
+
+			// This removes the need for the omitempty tag, to make the field non required.
+			// Required fields should be provided through a CustomSchemaApplier (https://pkg.go.dev/github.com/a-h/rest#CustomSchemaApplier)
+			//hasOmitEmptySet := slices.Contains(jsonTags, "omitempty")
```
Comment (jafhueppe) [inline schema.go:522]: nitpick: remove comment

**Review by jafhueppe — APPROVED** [review]
👍

File: schema.go (hunk 3 above)
Comment (clemenskohlhs) [reply schema.go]: Yes, the thought was to never set the required field and to use the CustomSchemaApplier for setting required fields. Maybe removing the whole section makes this more visible

**Review by juicyarts — APPROVED** [review]
This repo does not yet have a test pipeline, so please make sure that the tests respect this change and are running locally at least for the time being.

Other than that lgtm

**Pull request merged**